src/ir/passmanager.cpp

```diff
@@ -123,7 +123,10 @@ bool PassManager::runPass(Pass* p,vector<string>& pArgs) {
   if (argc > 1) {
     p->initialize(argc,argv);
   }
-  delete[] argv;
```

Comment:
we should just do the following:

```python
std::vector<char*> argv(argc);
for (auto& pArg : pArgs) {
  argv[i] = pArg.c_str();
}
```
then you don't have to deal w/ any `delete` business. There really shouldn't be raw `new`'s and `free`'s in the code.

(you may have to do some business w/ `const_cast<>` in the code above but I think it should work)

Reply:
Lets just use the answer in this SO https://stackoverflow.com/questions/26032039/convert-vectorstring-into-char-c

Reply:
believe that's exactly my suggestion above :D